src/routes/Project/actions/TrackActionHandler.js

```diff
@@ -47,6 +48,16 @@ const ACTION_HANDLERS = {
     let isMovingRight = oldClip.startTime < newClip.startTime;
     let isMovingLeft = !isMovingRight;
 
+
+    // grid snapping
```

Comment:
this seems like it can be done in constant time. 

interval = newClip.startTime/intervalWidth 
remainder = interval % 1
interval = remainder < ProjectConstants.GRID_SNAP_THRESHOLD ? interval : interval + 1
snapTarget = interval * intervalWidth